runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -121,8 +123,9 @@ void openCalculator(CalculatorType calculatorType)
 		// Update internal skill/XP values.
 		currentXP = client.getSkillExperience(calculatorType.getSkill());
 		currentLevel = Experience.getLevelForXp(currentXP);
-		targetLevel = enforceSkillBounds(currentLevel + 1);
-		targetXP = Experience.getXpForLevel(targetLevel);
+		VarPlayer endGoal = endGoalVarpForSkill(calculatorType.getSkill());
+		targetXP = endGoal != null ? client.getVar(endGoal) : 0;
+		targetLevel = enforceSkillBounds(Experience.getLevelForXp(targetXP));
```

Comment:
`getLevelForXp` always returns a level bound between the limits of `enforceSkillBounds`

Reply:
Removed the `enforceSkillBounds` call. Note: I didn't add a `enforceXPBounds` call around the `targetXP = client.getVar(endGoal);` line since RuneScape automatically enforces the 200M XP cap within the XP tracker.